app/src/sync/templates.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * templates.ts
+ * This module is responsible for fetching available templates from the server.
+ */
+
+import {GetListTemplatesResponse} from '@faims3/data-model';
+import {getTokenForCluster} from '../users';
+import {ListingsObject} from './databases';
+
+/**
+ * Fetches the list of templates from the server for a specific cluster.
+ *
+ * @param listing - Contains server information, including the cluster's unique ID and conductor URL.
+ * @returns A promise that resolves to an array of templates if the request is successful, or undefined if an error occurs.
+ */
+export const fetchTemplates = async (
+  listing: ListingsObject
+): Promise<GetListTemplatesResponse | undefined> => {
+  try {
+    // Retrieve a JWT token for the given cluster ID to authenticate the request
+    const jwt_token = await getTokenForCluster(listing._id);
+
+    // If no token is available, log an error and exit
+    if (!jwt_token) {
+      console.error('No token available for this cluster.');
+      return undefined;
```

Comment:
Not a fan of the 'return undefined when something goes wrong' pattern - what if the fetch doesn't have an item - it might return undefined too. Should clearly distinguish between the different possible responses and throw appropriate errors. 

Also if you wrap this async function in something like react-query, by obscuring the error you will get a pretty broad return type in the data field, and the error trigger will not occur.